src/carousel.tsx

```diff
@@ -457,6 +465,20 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
 
     const shouldAutoplay = useShouldAutoplay(!!autoplay, carouselRef);
 
+    React.useEffect(() => {
+        if (
+            initialActiveItem &&
+            (initialActiveItem !== initialItemInfo.current.initialItem ||
+                itemsPerPageFloor !== initialItemInfo.current.itemsPerPage ||
+                itemScrollPositions.length !== initialItemInfo.current.scrollPositionsLength)
+        ) {
+            goToPage(Math.floor(initialActiveItem / itemsPerPageFloor));
+            initialItemInfo.current.initialItem = initialActiveItem;
+            initialItemInfo.current.itemsPerPage = itemsPerPageFloor;
+            initialItemInfo.current.scrollPositionsLength = itemScrollPositions.length;
+        }
+    }, [initialActiveItem, itemsPerPageFloor, goToPage, itemScrollPositions.length]);
```

Comment:
I think this could be replaced by:
```
React.useEffect(() => {
    if (initialActiveItem !== undefined) {
        goToPage(Math.floor(initialActiveItem / itemsPerPageFloor));
    }
}, [initialActiveItem, goToPage, itemsPerPageFloor]);
```

But you also need to memoize the `pagesScrollPositions` value, because currently gets calculated in each render and `goToPage` callback depends on this value
```diff
-    const pagesScrollPositions = calcPagesScrollPositions(itemScrollPositions, pagesCount);
+    const pagesScrollPositions = React.useMemo(
+        () => calcPagesScrollPositions(itemScrollPositions, pagesCount),
+        [itemScrollPositions, pagesCount]
+    );
```

I've committed both changes to this branch `pladaria/carousel` and it seems to work as expected

Reply:
Will test this approach, coded something similar at the first attempt, and the carousel started to behave very bad.

Reply:
It works perfectly, I think what was missing in my first approach was the useMemo